<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8" />
<meta http-equiv="X-UA-Compatible" content="IE=edge" />
<meta name="viewport" content="width=device-width, initial-scale=1" /><!-- Begin Jekyll SEO tag v2.7.1 -->
<title>Bioinformatics tools: Stacks | kinow</title>
<meta name="generator" content="Jekyll v3.9.0" />
<meta property="og:title" content="Bioinformatics tools: Stacks" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="It is the first post about bioinformatics tools, but I will try to post more about other tools such as MrBayes, Structure, maybe some next generation sequencing tools too, and Bioperl, Biojava, and so on. As I am more a computer geek, rather than a bioinformatics one, I will focus on requirements for running these tools on clusters and the requirements to install them on your machine. The instructions require that you have an intermediary knowledge on *nix OS and sometimes a bit of programming experience.&lt;/em&gt; I will be using tutorials available on the Internet and hosting my code in GitHub/kinow. Hammer time!" />
<meta property="og:description" content="It is the first post about bioinformatics tools, but I will try to post more about other tools such as MrBayes, Structure, maybe some next generation sequencing tools too, and Bioperl, Biojava, and so on. As I am more a computer geek, rather than a bioinformatics one, I will focus on requirements for running these tools on clusters and the requirements to install them on your machine. The instructions require that you have an intermediary knowledge on *nix OS and sometimes a bit of programming experience.&lt;/em&gt; I will be using tutorials available on the Internet and hosting my code in GitHub/kinow. Hammer time!" />
<link rel="canonical" href="https://kinoshita.eti.br/2012/09/25/bioinformatic-tools-series-stacks.html" />
<meta property="og:url" content="https://kinoshita.eti.br/2012/09/25/bioinformatic-tools-series-stacks.html" />
<meta property="og:site_name" content="kinow" />
<meta property="og:type" content="article" />
<meta property="article:published_time" content="2012-09-25T00:00:00+12:00" />
<meta name="twitter:card" content="summary" />
<meta property="twitter:title" content="Bioinformatics tools: Stacks" />
<script type="application/ld+json">
{"description":"It is the first post about bioinformatics tools, but I will try to post more about other tools such as MrBayes, Structure, maybe some next generation sequencing tools too, and Bioperl, Biojava, and so on. As I am more a computer geek, rather than a bioinformatics one, I will focus on requirements for running these tools on clusters and the requirements to install them on your machine. The instructions require that you have an intermediary knowledge on *nix OS and sometimes a bit of programming experience.&lt;/em&gt; I will be using tutorials available on the Internet and hosting my code in GitHub/kinow. Hammer time!","headline":"Bioinformatics tools: Stacks","dateModified":"2012-09-25T00:00:00+12:00","datePublished":"2012-09-25T00:00:00+12:00","@type":"BlogPosting","mainEntityOfPage":{"@type":"WebPage","@id":"https://kinoshita.eti.br/2012/09/25/bioinformatic-tools-series-stacks.html"},"url":"https://kinoshita.eti.br/2012/09/25/bioinformatic-tools-series-stacks.html","@context":"https://schema.org"}</script>
<!-- End Jekyll SEO tag -->
<link rel="stylesheet" href="/assets/css/main.css" /><link type="application/atom+xml" rel="alternate" href="https://kinoshita.eti.br/feed.xml" title="kinow" /><link href="https://unpkg.com/pattern.css" rel="stylesheet" />

</head>
<body>
<div id="mobile-menu"><div class="menu">
  <ul class="menu">
    <li class="item">
      <h1 id="kinow">KINOW</h1>
    </li>
    <li class="item">
      <a href="/" class="">About</a>
    </li>
    <li class="item">
      <a href="/blog/" class="current">Blog</a>
    </li>
    <li class="item">
      <a href="/portfolio/" class="">Portfolio</a>
    </li>
    <li class="item social-media-item">
      <a href="https://www.instagram.com/brunokinoshita/" class="inline-link">
        <img src="/assets/icons/instagram.png" class="inverted-image" />
      </a>
      <a href="https://twitter.com/kinow/" class="inline-link">
        <img src="/assets/icons/twitter.png" class="inverted-image" />
      </a>
      <a href="https://github.com/kinow/" class="inline-link">
        <img src="/assets/icons/github.png" class="inverted-image" />
      </a>
    </li>
  </ul>
</div>
</div>
<div id="wrapper">
  <div id="sidebar"><div class="menu">
  <ul class="menu">
    <li class="item">
      <h1 id="kinow">KINOW</h1>
    </li>
    <li class="item">
      <a href="/" class="">About</a>
    </li>
    <li class="item">
      <a href="/blog/" class="current">Blog</a>
    </li>
    <li class="item">
      <a href="/portfolio/" class="">Portfolio</a>
    </li>
    <li class="item social-media-item">
      <a href="https://www.instagram.com/brunokinoshita/" class="inline-link">
        <img src="/assets/icons/instagram.png" class="inverted-image" />
      </a>
      <a href="https://twitter.com/kinow/" class="inline-link">
        <img src="/assets/icons/twitter.png" class="inverted-image" />
      </a>
      <a href="https://github.com/kinow/" class="inline-link">
        <img src="/assets/icons/github.png" class="inverted-image" />
      </a>
    </li>
  </ul>
</div>
</div>
  <div id="content">
    <article class="post" itemscope itemtype="http://schema.org/BlogPosting">
  <header class="post-header">
    <h1 class="post-title p-name" itemprop="name headline">Bioinformatics tools: Stacks</h1>
    <span class="post-meta">
      <time class="dt-published" datetime="2012-09-25T00:00:00+12:00" itemprop="datePublished">Sep 25, 2012
      </time>
      <span>&mdash; star date: -310185.79.</span>
      <span>
        Number of words: 1548.
      </span></span>
  </header>

  <ul>
  <li><a href="#installing-on-your-local-computer">Installing on your local computer</a></li>
  <li><a href="#installing-on-your-cluster">Installing on your cluster</a></li>
  <li><a href="#following-the-stacks-tutorial">Following the Stacks tutorial</a></li>
  <li><a href="#cpu-memory-and-disk-usage">CPU, memory and disk usage</a>
    <ul>
      <li><a href="#cpu-usage">CPU usage</a></li>
      <li><a href="#memory-usage">Memory usage</a></li>
      <li><a href="#disk-usage">Disk usage</a>
        <ul>
          <li><a href="#references">References</a></li>
        </ul>
      </li>
    </ul>
  </li>
</ul>


  <div class="post-content e-content" itemprop="articleBody">
    <p>It is the first post about <strong>bioinformatics tools</strong>, but I will try to post more about other tools such
as MrBayes, Structure, maybe some next generation sequencing tools too, and Bioperl, Biojava, and so on.</p>

<p>As I am more a computer geek, rather than a bioinformatics one, I will focus on requirements for running these tools
on clusters and the requirements to install them on your machine. The instructions require that you have an
intermediary knowledge on *nix OS and sometimes a bit of programming experience.&lt;/em&gt;</p>

<p><em>I will be using tutorials available on the Internet and hosting my code in <a href="http://www.github.com/kinow" alt="GitHub kinow" title="GitHub kinow">GitHub/kinow</a>. Hammer time!</em></p>

<!--more-->

<p><img class="ui fluid image" src="/assets/posts/2012-09-25-bioinformatics-tools-stacks/stacks_logo_300_156.png" /></p>

<p><strong>Stacks</strong> is hosted at <a href="http://www.uoregon.edu/" alt="Oregon University" title="Oregon University">Oregon University</a>, so if you are googling about it right now, probably a query including both ‘stacks’ and ‘oregon’ may give you better results (it’s good to know when you forget the web address). This is the stacks homepage: <a href="http://creskolab.uoregon.edu/stacks/">http://creskolab.uoregon.edu/stacks/</a>.</p>

<blockquote cite="http://creskolab.uoregon.edu/stacks/">"Stacks is a software pipeline for building loci out of a set of short-read sequenced samples. Stacks was developed for the purpose of building genetic maps from RAD-Tag Illumina sequence data, but can also be readily applied to population studies, and phylogeography."</blockquote>

<p>I will follow the tutorial <a href="http://creskolab.uoregon.edu/stacks/pe_tut.php" alt="Stacks tutorial" title="Stacks tutorial">“building mini-contigs from paired-end sequences”</a> to demonstrate how to use Stacks, and for each step executed I’ll post the directory and database sizes, as well as the processing time. My hardware is an Intel i5 quad core 2.30 GHz with 6GB of memory, running Debian 6.</p>

<!--more-->

<h2 id="installing-on-your-local-computer">Installing on your local computer</h2>

<p>The installation is quite simple. However, Stacks come with a Web application, and during its pipeline analysis sometimes it uploads data to a database. So in our set up I’m going to use an Apache HTTP web server and MySQL database. Here’s a list of things you should have installed in your machine:</p>

<ul>
<li><a href="http://httpd.apache.org" title="Apache httpd">Apache HTTP web server</a> (you could try NGINX, or others too, but you would need a PHP interpreter) - my version: Apache/2.2.22 (Debian)</li>
<li><a href="http://www.php.net" title="PHP">PHP</a> (not sure if 5&gt; is required) - my version: Ver 14.14 Distrib 5.5.24, for debian-linux-gnu (x86_64) using readline 6.2</li>
<li>PHP MDB2 module installed (pear install MDB2 does the trick most of the times)</li>
<li><a href="http://www.mysql.org/" title="MySQL">MySQL</a> database (stacks come with a SQL script for MySQL, not sure how hard it would be to adapt this script and the perl utilities to use Oracle, DB2, etc) - my version: Ver 14.14 Distrib 5.5.24, for debian-linux-gnu (x86_64) using readline 6.2</li>
<li><a href="http://www.perl.org" title="Perl">Perl</a> (5 is fine, didn't test with 6) - This is perl 5, version 14, subversion 2 (v5.14.2) built for x86_64-linux-gnu-thread-multi</li>
<li>DBD::mysql and SPREADSHEET::WriteExcel modules (<a href="http://www.cpan.org" title="CPAN">CPAN</a>, yay!)</li>
<li><a href="http://www.ebi.ac.uk/~zerbino/velvet/" title="Velvet">Velvet</a> (for contigs) - my version: 1.2.07</li>
<li>A gcc compiler with OpenMP for multi-threading. My version: 
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/4.7/lto-wrapper
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Debian 4.7.1-8' --with-bugurl=file:///usr/share/doc/gcc-4.7/README.Bugs --enable-languages=c,c++,go,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-4.7 --enable-shared --enable-linker-build-id --with-system-zlib --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --with-gxx-include-dir=/usr/include/c++/4.7 --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --enable-gnu-unique-object --enable-plugin --enable-objc-gc --with-arch-32=i586 --with-tune=generic --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 4.7.1 (Debian 4.7.1-8)</li>
</ul>

<p>Install Stacks with the classic <code class="language-plaintext highlighter-rouge">./configure</code>, <code class="language-plaintext highlighter-rouge">make</code>, <code class="language-plaintext highlighter-rouge">make install</code>. It will put several utilities and Perl scripts in your PATH. For testing if the installation worked, try running <code class="language-plaintext highlighter-rouge">ustacks</code>. You may need to update your Apache HTTP settings to include a virtual directory to the stacks web folder. In my computer this folder is located at <code class="language-plaintext highlighter-rouge">/usr/local/share/stacks/php</code>.</p>

<h2 id="installing-on-your-cluster">Installing on your cluster</h2>

<p>Stacks uses <strong>OpenMP</strong> for running over multiple CPU’s. I have a small commodity-hardware cluster that is quite handy for MapReduce and MPI processing, but for this case I will have to use only one machine. So I’m using my notebook with four cores, and using <a href="http://ganglia.sourceforge.net" title="Ganglia">Ganglia</a> and htop/free for monitoring CPU and memory usage.</p>

<h2 id="following-the-stacks-tutorial">Following the Stacks tutorial</h2>

<p>The tutorial requires that you download samples and create some directories and a database. Here’s the status of my system before the analysis.</p>

<p><strong>Database size: 880KB</strong>
<strong>Directory size: 2.6 GB</strong>
<code>kinow@chuva:~/Desktop/pe_tut$ du -h
4.0K	./stacks
4.0K	./assembled
2.6G	./samples
4.0K	./raw
4.0K	./pairs
2.6G	.</code></p>

<p>After these steps, you start your pipeline with the <code class="language-plaintext highlighter-rouge">denovo_map.pl</code> script. The denovo_map.pl script took 23 minutes to finish, and here’s the new status of my system.</p>

<p><strong>Database size: 1.7 GB</strong>
<strong>Directory size: 3.5 GB</strong>
<code>kinow@chuva:~/Desktop/pe_tut$ du -h
971M	./stacks
4.0K	./assembled
2.6G	./samples
4.0K	./raw
4.0K	./pairs
3.5G	.</code></p>

<p>The next step is the <code class="language-plaintext highlighter-rouge">export_sql.pl</code> script. It took 31 minutes to finish.</p>

<p><strong>Database size: 1.7 GB</strong>
<strong>Directory size: 3.5 GB</strong>
<code>kinow@chuva:~/Desktop/pe_tut$ du -h
971M	./stacks
4.0K	./assembled
2.6G	./samples
4.0K	./raw
4.0K	./pairs
3.5G	.</code></p>

<p>Half way to go. Now execute the <code class="language-plaintext highlighter-rouge">sort_read_pairs.pl</code> script. This was the longest and the one that consumed more memory, it took 49 minutes to finish.</p>

<p><strong>Database size: 1.7 GB</strong>
<strong>Directory size: 3.7 GB</strong>
<code>kinow@chuva:~/Desktop/pe_tut$ du -h
971M	./stacks
4.0K	./assembled
2.6G	./samples
4.0K	./raw
170M	./pairs
3.7G	.</code></p>

<p>It’s time to get the contigs with Velvet now with <code class="language-plaintext highlighter-rouge">exec_velvet.pl</code>. Don’t worry, this one is faster, only 15 minutes.</p>

<p><strong>Database size: 1.7 GB</strong>
<strong>Directory size: 3.7 GB</strong>
<code>kinow@chuva:~/Desktop/pe_tut$ du -h
971M	./stacks
3.4M	./assembled
2.6G	./samples
4.0K	./raw
170M	./pairs
3.7G	.</code></p>

<p>Last two scripts to execute. First <code class="language-plaintext highlighter-rouge">load_sequences.pl</code>, that apparently loads some data into the database. It took 3 minutes to finish.</p>

<p><strong>Database size: 1.8 GB</strong>
<strong>Directory size: 3.7 GB</strong>
<code>kinow@chuva:~/Desktop/pe_tut$ du -h
971M	./stacks
3.4M	./assembled
2.6G	./samples
4.0K	./raw
170M	./pairs
3.7G	.</code></p>

<p>And finally <code class="language-plaintext highlighter-rouge">index_radtags.pl</code> which took only 1 minute.</p>

<p><strong>Database size: 1.8 GB</strong>
<strong>Directory size: 3.7 GB</strong>
<code>kinow@chuva:~/Desktop/pe_tut$ du -h
971M	./stacks
3.4M	./assembled
2.6G	./samples
4.0K	./raw
170M	./pairs
3.7G	.</code></p>

<h2 id="cpu-memory-and-disk-usage">CPU, memory and disk usage</h2>

<p>I was using Ganglia for monitoring Hadoop jobs, so I saved the graphs for the timeframe of the pipeline execution. I started to execute this analysis at 10AM, and finished about 2 and a half hours later (the process is not automated, so sometimes I had to check the output and documentation before executing the next step).</p>

<p>Following you’ll find the raw graphs and another version with legends to help understand what was being executed.</p>

<h3 id="cpu-usage">CPU usage</h3>

<p><img class="ui fluid image" src="/assets/posts/2012-09-25-bioinformatics-tools-stacks/cpu.png" /></p>

<p><img class="ui fluid image" src="/assets/posts/2012-09-25-bioinformatics-tools-stacks/cpu_legends.png" /></p>

<h3 id="memory-usage">Memory usage</h3>

<p><img class="ui fluid image" src="/assets/posts/2012-09-25-bioinformatics-tools-stacks/memory.png" /></p>

<p><img class="ui fluid image" src="/assets/posts/2012-09-25-bioinformatics-tools-stacks/memory_legends.png" /></p>

<h3 id="disk-usage">Disk usage</h3>

<p>I thought the disk graphs from Ganglia wouldn’t represent the disk usage very well, specially since my disk has about 500GB, so it’s harder to see the changes. So I used some R to plot a graph that I hope can demonstrate the usage for this tutorial. <strong>It is important to highlight that we are using samples from a tutorial, and your analysis may produce high or lower disk usage.</strong></p>

<p><img class="ui fluid image" src="/assets/posts/2012-09-25-bioinformatics-tools-stacks/disk.png" /></p>

<p>Hope it helps you using Stacks. Kudos to all developers and contributors of this great tool.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">disk</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">2.6</span><span class="p">,</span><span class="m">3.5</span><span class="p">,</span><span class="m">3.5</span><span class="p">,</span><span class="m">3.7</span><span class="p">,</span><span class="m">3.7</span><span class="p">,</span><span class="m">3.7</span><span class="p">,</span><span class="m">3.7</span><span class="p">)</span><span class="w">
</span><span class="n">db</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">0.0008</span><span class="p">,</span><span class="m">1.7</span><span class="p">,</span><span class="m">1.7</span><span class="p">,</span><span class="m">1.7</span><span class="p">,</span><span class="m">1.7</span><span class="p">,</span><span class="m">1.8</span><span class="p">,</span><span class="m">1.8</span><span class="p">)</span><span class="w">
</span><span class="n">x</span><span class="o">=</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">7</span><span class="p">)</span><span class="w">
</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="n">xlab</span><span class="o">=</span><span class="s2">"Tutorial Step (1 to 7)"</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="n">disk</span><span class="p">,</span><span class="w"> </span><span class="n">ylab</span><span class="o">=</span><span class="s2">"Disk Usage (GB)"</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="s2">"red"</span><span class="p">,</span><span class="w"> </span><span class="n">type</span><span class="o">=</span><span class="s2">"o"</span><span class="p">,</span><span class="n">ylim</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">6</span><span class="p">),</span><span class="n">xlim</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">8</span><span class="p">))</span><span class="w">
</span><span class="n">lines</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="n">db</span><span class="p">,</span><span class="w"> </span><span class="n">type</span><span class="o">=</span><span class="s2">"o"</span><span class="p">,</span><span class="w"> </span><span class="n">pch</span><span class="o">=</span><span class="m">22</span><span class="p">,</span><span class="w"> </span><span class="n">lty</span><span class="o">=</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="s2">"blue"</span><span class="p">)</span><span class="w">
</span><span class="n">title</span><span class="p">(</span><span class="n">main</span><span class="o">=</span><span class="s2">"Disk Usage"</span><span class="p">,</span><span class="w"> </span><span class="n">font.main</span><span class="o">=</span><span class="m">2</span><span class="p">)</span><span class="w">
</span><span class="n">abline</span><span class="p">(</span><span class="n">v</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="o">:</span><span class="m">8</span><span class="p">),</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="s2">"grey40"</span><span class="p">,</span><span class="w"> </span><span class="n">lty</span><span class="o">=</span><span class="s2">"dotted"</span><span class="p">)</span><span class="w">
</span><span class="n">abline</span><span class="p">(</span><span class="n">h</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="o">:</span><span class="m">5</span><span class="p">),</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="s2">"grey10"</span><span class="p">,</span><span class="w"> </span><span class="n">lty</span><span class="o">=</span><span class="s2">"dotted"</span><span class="p">)</span><span class="w">
</span><span class="n">df</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">disk</span><span class="p">,</span><span class="n">db</span><span class="p">)</span><span class="w">
</span><span class="n">totals</span><span class="o">=</span><span class="n">rowSums</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="n">na.rm</span><span class="o">=</span><span class="nb">T</span><span class="p">)</span><span class="w">
</span><span class="n">lines</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="n">totals</span><span class="p">,</span><span class="w"> </span><span class="n">type</span><span class="o">=</span><span class="s2">"o"</span><span class="p">,</span><span class="w"> </span><span class="n">pch</span><span class="o">=</span><span class="m">22</span><span class="p">,</span><span class="w"> </span><span class="n">lty</span><span class="o">=</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="s2">"green"</span><span class="p">)</span><span class="w"> 
</span><span class="n">legend</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">6</span><span class="p">,</span><span class="nf">c</span><span class="p">(</span><span class="s2">"DB"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Disk"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Total"</span><span class="p">),</span><span class="w"> </span><span class="n">cex</span><span class="o">=</span><span class="m">0.6</span><span class="p">,</span><span class="n">col</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s2">"red"</span><span class="p">,</span><span class="w"> </span><span class="s2">"blue"</span><span class="p">,</span><span class="w"> </span><span class="s2">"green"</span><span class="p">),</span><span class="n">pch</span><span class="o">=</span><span class="m">21</span><span class="o">:</span><span class="m">23</span><span class="p">,</span><span class="n">lty</span><span class="o">=</span><span class="m">1</span><span class="p">,</span><span class="n">y.intersp</span><span class="o">=</span><span class="m">0.2</span><span class="p">,</span><span class="n">x.intersp</span><span class="o">=</span><span class="m">0.2</span><span class="p">,</span><span class="n">pt.lwd</span><span class="o">=</span><span class="m">1</span><span class="p">,</span><span class="n">adj</span><span class="o">=</span><span class="m">0</span><span class="p">,</span><span class="n">inset</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">0.5</span><span class="p">,</span><span class="m">0.5</span><span class="p">))</span><span class="w">
</span></code></pre></div></div>

<h4 id="references">References</h4>

<ul>
<li>Stacks - <a href="http://creskolab.uoregon.edu/stacks/">http://creskolab.uoregon.edu/stacks/</a></li>
<li>Tutorial: building mini-contigs from paired-end sequences - <a href="http://creskolab.uoregon.edu/stacks/pe_tut.php">http://creskolab.uoregon.edu/stacks/pe_tut.php</a></li>
</ul>

  </div>

  <div class="tags"><p>Tags:&nbsp;<a class="label" href="/tag/bioinformatics">bioinformatics</a>.<br/></p></div>

  <a class="u-url" href="/2012/09/25/bioinformatic-tools-series-stacks.html" hidden></a>
</article>

  </div>
</div>
</body>
</html>
