<!doctype html><html lang=en>
<head>
<meta charset=utf-8>
<meta name=viewport content="width=device-width,initial-scale=1">
<meta property="og:title" content="Bioinformatics tools: Stacks">
<meta property="og:description" content="It is the first post about bioinformatics tools, but I will try to post more about other tools such
as MrBayes, Structure, maybe some next generation sequencing tools too, and Bioperl, Biojava, and so on.
As I am more a computer geek, rather than a bioinformatics one, I will focus on requirements for running these tools
on clusters and the requirements to install them on your machine. The instructions require that you have an
intermediary knowledge on *nix OS and sometimes a bit of programming experience.
I will be using tutorials available on the Internet and hosting my code in GitHub/kinow. Hammer time!">
<meta property="og:type" content="article">
<meta property="og:url" content="https://kinoshita.eti.br/2012/09/25/bioinformatics-tools-stacks.html"><meta property="article:section" content="posts">
<meta property="article:published_time" content="2012-09-25T00:00:00+00:00">
<meta property="article:modified_time" content="2012-09-25T00:00:00+00:00">
<link rel=stylesheet type=text/css href=https://kinoshita.eti.br/sass/main.66aa172f782442adc0241c99968a79a72e4260187c78830f9bdd6b03abff6103.css integrity="sha256-ZqoXL3gkQq3AJByZlop5py5CYBh8eIMPm91rA6v/YQM=">
<title>kinow | Bioinformatics tools: Stacks</title>
</head>
<body>
<header role=banner class=content>
<nav id=nav aria-label=Main role=navigation class=content>
<ul><li>
<a href=/><i data-feather=about></i> About</a>
</li><li>
<a href=/blog/><i data-feather=blog></i> Blog</a>
</li><li>
<a href=/portfolio/><i data-feather=portfolio></i> Portfolio</a>
</li>
</ul>
</nav>
</header>
<main>
<article class=content>
<h1>Bioinformatics tools: Stacks</h1><div class=metadata>
<i data-feather=calendar></i>
<time datetime=2012-09-25>Sep 25, 2012</time>
</div>
<aside>
<nav id=TableOfContents>
<ul>
<li><a href=#installing-on-your-local-computer>Installing on your local computer</a></li>
<li><a href=#installing-on-your-cluster>Installing on your cluster</a></li>
<li><a href=#following-the-stacks-tutorial>Following the Stacks tutorial</a></li>
<li><a href=#cpu-memory-and-disk-usage>CPU, memory and disk usage</a>
<ul>
<li><a href=#cpu-usage>CPU usage</a></li>
<li><a href=#memory-usage>Memory usage</a></li>
<li><a href=#disk-usage>Disk usage</a></li>
</ul>
</li>
</ul>
</nav>
</aside>
<p>It is the first post about <strong>bioinformatics tools</strong>, but I will try to post more about other tools such
as MrBayes, Structure, maybe some next generation sequencing tools too, and Bioperl, Biojava, and so on.</p>
<p>As I am more a computer geek, rather than a bioinformatics one, I will focus on requirements for running these tools
on clusters and the requirements to install them on your machine. The instructions require that you have an
intermediary knowledge on *nix OS and sometimes a bit of programming experience.</em></p>
<p><em>I will be using tutorials available on the Internet and hosting my code in <a href=http://www.github.com/kinow alt="GitHub kinow" title="GitHub kinow">GitHub/kinow</a>. Hammer time!</p>
<figure class=feature>
<img src=/assets/posts/2012-09-25-bioinformatic-tools-series-stacks/stacks_logo_300_156.png alt title width height>
<figcaption></figcaption>
</figure>
<p><strong>Stacks</strong> is hosted at <a href=http://www.uoregon.edu/ alt="Oregon University" title="Oregon University">Oregon University</a>, so if you are googling about it right now, probably a query including both &lsquo;stacks&rsquo; and &lsquo;oregon&rsquo; may give you better results (it&rsquo;s good to know when you forget the web address). This is the stacks homepage: <a href=http://creskolab.uoregon.edu/stacks/><a href=http://creskolab.uoregon.edu/stacks/>http://creskolab.uoregon.edu/stacks/</a></a>.</p>
<blockquote cite=http://creskolab.uoregon.edu/stacks/>"Stacks is a software pipeline for building loci out of a set of short-read sequenced samples. Stacks was developed for the purpose of building genetic maps from RAD-Tag Illumina sequence data, but can also be readily applied to population studies, and phylogeography."</blockquote>
<p>I will follow the tutorial <a href=http://creskolab.uoregon.edu/stacks/pe_tut.php alt="Stacks tutorial" title="Stacks tutorial">&ldquo;building mini-contigs from paired-end sequences&rdquo;</a> to demonstrate how to use Stacks, and for each step executed I&rsquo;ll post the directory and database sizes, as well as the processing time. My hardware is an Intel i5 quad core 2.30 GHz with 6GB of memory, running Debian 6.</p>
<h2 id=installing-on-your-local-computer>Installing on your local computer</h2>
<p>The installation is quite simple. However, Stacks come with a Web application, and during its pipeline analysis sometimes it uploads data to a database. So in our set up I&rsquo;m going to use an Apache HTTP web server and MySQL database. Here&rsquo;s a list of things you should have installed in your machine:</p>
<ul>
<li><a href=http://httpd.apache.org title="Apache httpd">Apache HTTP web server</a> (you could try NGINX, or others too, but you would need a PHP interpreter) - my version: Apache/2.2.22 (Debian)</li>
<li><a href=http://www.php.net title=PHP>PHP</a> (not sure if 5> is required) - my version: Ver 14.14 Distrib 5.5.24, for debian-linux-gnu (x86_64) using readline 6.2</li>
<li>PHP MDB2 module installed (pear install MDB2 does the trick most of the times)</li>
<li><a href=http://www.mysql.org/ title=MySQL>MySQL</a> database (stacks come with a SQL script for MySQL, not sure how hard it would be to adapt this script and the perl utilities to use Oracle, DB2, etc) - my version: Ver 14.14 Distrib 5.5.24, for debian-linux-gnu (x86_64) using readline 6.2</li>
<li><a href=http://www.perl.org title=Perl>Perl</a> (5 is fine, didn't test with 6) - This is perl 5, version 14, subversion 2 (v5.14.2) built for x86_64-linux-gnu-thread-multi</li>
<li>DBD::mysql and SPREADSHEET::WriteExcel modules (<a href=http://www.cpan.org title=CPAN>CPAN</a>, yay!)</li>
<li><a href=http://www.ebi.ac.uk/~zerbino/velvet/ title=Velvet>Velvet</a> (for contigs) - my version: 1.2.07</li>
<li>A gcc compiler with OpenMP for multi-threading. My version:
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/4.7/lto-wrapper
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Debian 4.7.1-8' --with-bugurl=file:///usr/share/doc/gcc-4.7/README.Bugs --enable-languages=c,c++,go,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-4.7 --enable-shared --enable-linker-build-id --with-system-zlib --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --with-gxx-include-dir=/usr/include/c++/4.7 --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --enable-gnu-unique-object --enable-plugin --enable-objc-gc --with-arch-32=i586 --with-tune=generic --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 4.7.1 (Debian 4.7.1-8)</li>
</ul>
<p>Install Stacks with the classic <code>./configure</code>, <code>make</code>, <code>make install</code>. It will put several utilities and Perl scripts in your PATH. For testing if the installation worked, try running <code>ustacks</code>. You may need to update your Apache HTTP settings to include a virtual directory to the stacks web folder. In my computer this folder is located at <code>/usr/local/share/stacks/php</code>.</p>
<h2 id=installing-on-your-cluster>Installing on your cluster</h2>
<p>Stacks uses <strong>OpenMP</strong> for running over multiple CPU&rsquo;s. I have a small commodity-hardware cluster that is quite handy for MapReduce and MPI processing, but for this case I will have to use only one machine. So I&rsquo;m using my notebook with four cores, and using <a href=http://ganglia.sourceforge.net title=Ganglia>Ganglia</a> and htop/free for monitoring CPU and memory usage.</p>
<h2 id=following-the-stacks-tutorial>Following the Stacks tutorial</h2>
<p>The tutorial requires that you download samples and create some directories and a database. Here&rsquo;s the status of my system before the analysis.</p>
<p><strong>Database size: 880KB</strong>
<strong>Directory size: 2.6 GB</strong>
<code>kinow@chuva:~/Desktop/pe_tut$ du -h
4.0K ./stacks
4.0K ./assembled
2.6G ./samples
4.0K ./raw
4.0K ./pairs
2.6G .</code></p>
<p>After these steps, you start your pipeline with the <code>denovo_map.pl</code> script. The denovo_map.pl script took 23 minutes to finish, and here&rsquo;s the new status of my system.</p>
<p><strong>Database size: 1.7 GB</strong>
<strong>Directory size: 3.5 GB</strong>
<code>kinow@chuva:~/Desktop/pe_tut$ du -h
971M ./stacks
4.0K ./assembled
2.6G ./samples
4.0K ./raw
4.0K ./pairs
3.5G .</code></p>
<p>The next step is the <code>export_sql.pl</code> script. It took 31 minutes to finish.</p>
<p><strong>Database size: 1.7 GB</strong>
<strong>Directory size: 3.5 GB</strong>
<code>kinow@chuva:~/Desktop/pe_tut$ du -h
971M ./stacks
4.0K ./assembled
2.6G ./samples
4.0K ./raw
4.0K ./pairs
3.5G .</code></p>
<p>Half way to go. Now execute the <code>sort_read_pairs.pl</code> script. This was the longest and the one that consumed more memory, it took 49 minutes to finish.</p>
<p><strong>Database size: 1.7 GB</strong>
<strong>Directory size: 3.7 GB</strong>
<code>kinow@chuva:~/Desktop/pe_tut$ du -h
971M ./stacks
4.0K ./assembled
2.6G ./samples
4.0K ./raw
170M ./pairs
3.7G .</code></p>
<p>It&rsquo;s time to get the contigs with Velvet now with <code>exec_velvet.pl</code>. Don&rsquo;t worry, this one is faster, only 15 minutes.</p>
<p><strong>Database size: 1.7 GB</strong>
<strong>Directory size: 3.7 GB</strong>
<code>kinow@chuva:~/Desktop/pe_tut$ du -h
971M ./stacks
3.4M ./assembled
2.6G ./samples
4.0K ./raw
170M ./pairs
3.7G .</code></p>
<p>Last two scripts to execute. First <code>load_sequences.pl</code>, that apparently loads some data into the database. It took 3 minutes to finish.</p>
<p><strong>Database size: 1.8 GB</strong>
<strong>Directory size: 3.7 GB</strong>
<code>kinow@chuva:~/Desktop/pe_tut$ du -h
971M ./stacks
3.4M ./assembled
2.6G ./samples
4.0K ./raw
170M ./pairs
3.7G .</code></p>
<p>And finally <code>index_radtags.pl</code> which took only 1 minute.</p>
<p><strong>Database size: 1.8 GB</strong>
<strong>Directory size: 3.7 GB</strong>
<code>kinow@chuva:~/Desktop/pe_tut$ du -h
971M ./stacks
3.4M ./assembled
2.6G ./samples
4.0K ./raw
170M ./pairs
3.7G .</code></p>
<h2 id=cpu-memory-and-disk-usage>CPU, memory and disk usage</h2>
<p>I was using Ganglia for monitoring Hadoop jobs, so I saved the graphs for the timeframe of the pipeline execution. I started to execute this analysis at 10AM, and finished about 2 and a half hours later (the process is not automated, so sometimes I had to check the output and documentation before executing the next step).</p>
<p>Following you&rsquo;ll find the raw graphs and another version with legends to help understand what was being executed.</p>
<h3 id=cpu-usage>CPU usage</h3>
<figure class=feature>
<img src=/assets/posts/2012-09-25-bioinformatic-tools-series-stacks/cpu.png alt title width height>
<figcaption></figcaption>
</figure>
<figure class=feature>
<img src=/assets/posts/2012-09-25-bioinformatic-tools-series-stacks/cpu_legends.png alt title width height>
<figcaption></figcaption>
</figure>
<h3 id=memory-usage>Memory usage</h3>
<figure class=feature>
<img src=/assets/posts/2012-09-25-bioinformatic-tools-series-stacks/memory.png alt title width height>
<figcaption></figcaption>
</figure>
<figure class=feature>
<img src=/assets/posts/2012-09-25-bioinformatic-tools-series-stacks/memory_legends.png alt title width height>
<figcaption></figcaption>
</figure>
<h3 id=disk-usage>Disk usage</h3>
<p>I thought the disk graphs from Ganglia wouldn&rsquo;t represent the disk usage very well, specially since my disk has about 500GB, so it&rsquo;s harder to see the changes. So I used some R to plot a graph that I hope can demonstrate the usage for this tutorial. <strong>It is important to highlight that we are using samples from a tutorial, and your analysis may produce high or lower disk usage.</strong></p>
<figure class=feature>
<img src=/assets/posts/2012-09-25-bioinformatic-tools-series-stacks/disk.png alt title width height>
<figcaption></figcaption>
</figure>
<p>Hope it helps you using Stacks. Kudos to all developers and contributors of this great tool.</p>
<div class=highlight><pre tabindex=0 style=background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-r data-lang=r>disk=c(<span style=color:#00f>2.6</span>,<span style=color:#00f>3.5</span>,<span style=color:#00f>3.5</span>,<span style=color:#00f>3.7</span>,<span style=color:#00f>3.7</span>,<span style=color:#00f>3.7</span>,<span style=color:#00f>3.7</span>)
db=c(<span style=color:#00f>0.0008</span>,<span style=color:#00f>1.7</span>,<span style=color:#00f>1.7</span>,<span style=color:#00f>1.7</span>,<span style=color:#00f>1.7</span>,<span style=color:#00f>1.8</span>,<span style=color:#00f>1.8</span>)
x=(<span style=color:#00f>1</span>:<span style=color:#00f>7</span>)
plot(x=x, xlab=<span style=color:#00f>&#34;Tutorial Step (1 to 7)&#34;</span>, y=disk, ylab=<span style=color:#00f>&#34;Disk Usage (GB)&#34;</span>, col=<span style=color:#00f>&#34;red&#34;</span>, type=<span style=color:#00f>&#34;o&#34;</span>,ylim=c(<span style=color:#00f>0</span>,<span style=color:#00f>6</span>),xlim=c(<span style=color:#00f>0</span>,<span style=color:#00f>8</span>))
lines(x=x, y=db, type=<span style=color:#00f>&#34;o&#34;</span>, pch=<span style=color:#00f>22</span>, lty=<span style=color:#00f>2</span>, col=<span style=color:#00f>&#34;blue&#34;</span>)
title(main=<span style=color:#00f>&#34;Disk Usage&#34;</span>, font.main=<span style=color:#00f>2</span>)
abline(v=c(<span style=color:#00f>0</span>:<span style=color:#00f>8</span>), col=<span style=color:#00f>&#34;grey40&#34;</span>, lty=<span style=color:#00f>&#34;dotted&#34;</span>)
abline(h=c(<span style=color:#00f>0</span>:<span style=color:#00f>5</span>), col=<span style=color:#00f>&#34;grey10&#34;</span>, lty=<span style=color:#00f>&#34;dotted&#34;</span>)
df = data.frame(disk,db)
totals=rowSums(df,na.rm=T)
lines(x=x, y=totals, type=<span style=color:#00f>&#34;o&#34;</span>, pch=<span style=color:#00f>22</span>, lty=<span style=color:#00f>2</span>, col=<span style=color:#00f>&#34;green&#34;</span>) 
legend(<span style=color:#00f>0</span>,<span style=color:#00f>6</span>,c(<span style=color:#00f>&#34;DB&#34;</span>, <span style=color:#00f>&#34;Disk&#34;</span>, <span style=color:#00f>&#34;Total&#34;</span>), cex=<span style=color:#00f>0.6</span>,col=c(<span style=color:#00f>&#34;red&#34;</span>, <span style=color:#00f>&#34;blue&#34;</span>, <span style=color:#00f>&#34;green&#34;</span>),pch=<span style=color:#00f>21</span>:<span style=color:#00f>23</span>,lty=<span style=color:#00f>1</span>,y.intersp=<span style=color:#00f>0.2</span>,x.intersp=<span style=color:#00f>0.2</span>,pt.lwd=<span style=color:#00f>1</span>,adj=<span style=color:#00f>0</span>,inset=c(<span style=color:#00f>0.5</span>,<span style=color:#00f>0.5</span>))
</code></pre></div><h4 id=references>References</h4>
<ul>
<li>Stacks - <a href=http://creskolab.uoregon.edu/stacks/>http://creskolab.uoregon.edu/stacks/</a></li>
<li>Tutorial: building mini-contigs from paired-end sequences - <a href=http://creskolab.uoregon.edu/stacks/pe_tut.php>http://creskolab.uoregon.edu/stacks/pe_tut.php</a></li>
</ul>
<aside class=links>
<a class=previous href=https://kinoshita.eti.br/2012/09/25/paper-patterns-for-introducing-a-superclass-for-test-classes.html>&#171; Previous</a>
<a class=next href=https://kinoshita.eti.br/2012/10/11/jenkins-testlink-and-gtest-in-5-minutes-or-so.html>Next &#187;</a>
</aside>
</article>
</main>
<footer role=contentinfo>
<ul id=social-media-icons>
<li>
<a href=https://www.instagram.com/brunokinoshita/>
<img src=/assets/icons/instagram.png alt="Instagram link">
</a>
</li>
<li>
<a href=https://twitter.com/kinow/>
<img src=/assets/icons/twitter.png alt="Twitter link">
</a>
</li>
<li>
<a href=https://github.com/kinow/>
<img src=/assets/icons/github.png alt="GitHub link">
</a>
</li>
<li>
<a href=/feed.xml>
<img src=/assets/icons/news.png alt="RSS feed link">
</a>
</li>
</ul>
</footer>
</body>
</html>